// @aws-sdk/client-cloudwatch-logs/dist-es/commands/PutQueryDefinitionCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { PutQueryDefinitionRequestFilterSensitiveLog, PutQueryDefinitionResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1PutQueryDefinitionCommand, serializeAws_json1_1PutQueryDefinitionCommand, } from "../protocols/Aws_json1_1";
export class PutQueryDefinitionCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "CloudWatchLogsClient";
        const commandName = "PutQueryDefinitionCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: PutQueryDefinitionRequestFilterSensitiveLog,
            outputFilterSensitiveLog: PutQueryDefinitionResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_1PutQueryDefinitionCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_1PutQueryDefinitionCommand(output, context);
    }
}
